numpy/lib/tests/test_financial.py

```diff
@@ -344,3 +346,25 @@ def test_broadcast_decimal(self):
                                     Decimal('0'), [Decimal('0'), Decimal('0'), Decimal('1'), 'end', 'begin']),
                             [Decimal('-74.998201'), Decimal('-75.62318601'), Decimal('-75.62318601'),
                              Decimal('-76.88882405'), Decimal('-76.88882405')], 4)
+        
+    # Test for checking inputs whose output is NaN
+    @pytest.mark.parametrize('number_type', [Decimal, float])
+    def test_rate_nan(self, number_type):
+        # Rate will return NaN, if newton raphson method's change or diff was not able to become
+        # less than default tolerance value i.e. 1e-6 in max iterations possible,
+        rate = np.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(0))
+        assert_equal(np.nan, float(rate))
+        rate = np.rate(number_type(12.0), number_type('-400.0'), number_type('10000.0'), number_type(20000))
+        assert_equal(np.nan, float(rate))
+        
+        # begin
+        rate = np.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 1)
+        assert_equal(np.nan, float(rate))
+        rate = np.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(20000), 'begin')
+        assert_equal(np.nan, float(rate))
+        # end
+        
+        rate = np.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(0))
+        assert_equal(np.nan, float(rate))
+        rate = np.rate(number_type(12.0), number_type('400.0'), number_type('10000.0'), number_type(0), 'end')
+        assert_equal(np.nan, float(rate))  
```

Comment:
nit: i think this `end` comment is describing the `'end'` below, so the blank line should be before `#end` not after it